### PR TITLE
feat(design-tokens): update tokens and components for dark theme

### PR DESCRIPTION
**Pull request by loreina**
## paste-website
- Updated Sidebar right border to use `colorBorderWeak`

## paste-core
- Updated `PrimaryButton` and `DestructiveButton` default text color to `colorTextInverse`
 - Updated `LinkButton` active state to use `colorTextLinkStrongest`
- Updated `DestructiveLinkButton` active state to use `colorTextLinkDestructiveStrongest`
- Updated `InverseButton` to use `shadowBorderInverseWeakest` when disabled
- Updated `Checkbox` and `Radio` to use `colorTextInverse` when checked, and `ColorTextWeakest` when not, which are all updated in `BaseRadioCheckbox`
- ~~Swap `Radio` circle to use `RadioDotIcon` instead of a `Box`~~
- Updated `TooltipArrow` to use `colorBorderInverseWeaker` for stroke color
- Updated `Tooltip` to use `colorBorderInverseWeaker` as background color
- Updated `NeutralToast` to use `colorBorderNeutralWeak` for border color
- Updated neutral `Alert` to use `colorBorderNeutralWeak` for border color
- Updated `DisplayPill` anchor variant to use `colorText`
- Updated `DefaultFauxInput` to use `shadowBorderPrimaryStronger` in default and `shadowBorderErrorStronger` in error
- Updated Modal borders to use `colorBorderWeak`
- Updated `InverseFauxInput` to use `colorBackgroundInverse` as background color
- Checkbox and Radio updates
  - Unchecked
     - Hover and focus border -> $color-border-primary-stronger
  - Checked
     - Hover, active, and focus background -> $color-background-primary-stronger
     - Hover, active, and focus border -> $color-border-primary-stronger
     - Default icon -> $color-text-inverse
  - Unchecked error
     - Hover and focus border -> $color-border-error-stronger (NEW token)
  - Checked error
     - Hover, active, and focus background -> $color-background-error-stronger
     - Hover, active, and focus border -> $color-border-error-stronger
     - Default icon -> $color-text-inverse
     - Unchecked, disabled, hover, focus icon -> $color-text-weakest 

### Aliases
- Added a new alias `palette-gray-110`

### Global: background-color
- Updated `color-background-inverse` to gray-90
- Updated `color-background-inverse-strong` to gray-80

### Global: border color
- Added `color-border-inverse-weakest` token pointing to gray-90
- Updated `color-border-inverse-weaker` to gray-80
- Updated `color-border-inverse` to gray-50
- Updated `color-border-inverse-strong` to gray-20
- Updated `color-border-inverse-stronger` to gray-10
- Neutral
  - Updated `color-border-neutral-weak` to blue-40
- Added `color-border-error-stronger` token pointing to red-90

### Global: box shadow
- Added `shadow-border-error-stronger` pointing to red-90
- Added `shadow-border-inverse-weakest` pointing to gray-90
- Updated `shadow-border-inverse-weaker` to gray-80
- Updated `shadow-border-inverse` to gray-50
- - Updated `shadow-border-inverse-strong` to gray-20
- - Updated `shadow-border-inverse-stronger` to gray-10

### Dark theme: background color
- Updated `color-background` to gray-100
- Updated `color-background-strong` to gray-90
- Updated `color-background-stronger` to gray-80
- Updated `color-background-inverse` to gray-90
- Updated `color-background-inverse-strong` to gray-80
- Updated `color-background-body` to new alias gray-110
- Updated `color-background-overlay` to palette-gray-100-transparent-90
- Updated `color-background-row-striped` to gray-100
- Updated `color-background-subaccount` to gray-90
- Updated `color-background-trial` to green-100
- Brand
  - Updated `color-background-brand` to gray-100
  - Updated `color-background-brand-strong` to gray-90
- Primary
  - Updated `color-background-primary-strongest` to blue-10
  - Updated `color-background-primary-stronger` to blue-20
  - Updated `color-background-primary-strong` to blue-30
  - Updated `color-background-primary` to blue-60
  - Updated `color-background-primary-weak` to gray-80
  - Updated `color-background-primary-weaker` to gray-90
  - Updated `color-background-primary-weakest` to gray-100
- Destructive
  - Updated `color-background-destructive-strongest` to red-10
  - Updated `color-background-destructive-stronger` to red-20
  - Updated `color-background-destructive-strong` to red-30
  - Updated `color-background-destructive` to red-60
  - Updated `color-background-destructive-weak` to gray-80
  - Updated `color-background-destructive-weaker` to gray-90
  - Updated `color-background-destructive-weakest` to gray-100
- Success
  - Updated `color-background-success` to green-60
- Warning
  - Updated `color-background-warning` to orange-60
- Error
  - Added `color-background-error-stronger` pointing to red-20
  - Updated `color-background-error-strong` to red-40
  - Updated `color-background-error` to red-60

### Dark theme: border color
- Updated `color-border-weak` to gray-80
- Updated `color-border-weaker` to gray-90
- Inverse
  - Removed `color-border-inverse-weaker`
  - Removed `color-border-inverse`
  - Removed `color-border-inverse-strong`
  - Removed `color-border-inverse-stronger`
  - Removed `color-border-inverse-strongest`
- Primary
  - Updated `color-border-primary-strongest` to blue-10
  - Updated `color-border-primary-stronger` to blue-20
  - Updated `color-border-primary-strong` to blue-30
  - Updated `color-border-primary` to blue-60
  - Updated `color-border-primary-weak` to gray-80
  - Updated `color-border-primary-weaker` to gray-90
  - Updated `color-border-primary-weakest` to gray-100
- Destructive
  - Updated `color-border-destructive-strongest` to red-10
  - Updated `color-border-destructive-stronger` to red-20
  - Updated `color-border-destructive-strong` to red-30
  - Updated `color-border-destructive` to red-60
  - Updated `color-border-destructive-weak` to gray-80
  - Updated `color-border-destructive-weaker` to gray-90
  - Updated `color-border-destructive-weakest` to gray-100
- Success
  - Updated `color-border-success` to green-60
  - Updated `color-border-success-weak` to green-70
- Warning
  - Updated `color-border-warning` to orange-60
  - Updated `color-border-warning-weak` to orange-70
- Error
  - Added `color-border-error-stronger` pointing to red-20
  - Updated `color-border-error-strong` to red-30
  - Updated `color-border-error` to red-60
  - Updated `color-border-error-weak` to red-70
- Neutral
  - Updated `color-border-neutral` to blue-60
  - Updated `color-border-neutral-weak` to blue-70

### Dark theme: box shadow
- Elevation
  - Update `shadow-elevation-10` to `black-transparent-40`
  - Update `shadow-elevation-20` to `black-transparent-50`
  - Update `shadow-elevation-30` to `black-transparent-50`
- Focus
  - Updated `shadow-focus` to gray-60
  - Updated `shadow-focus-inverse` to gray-50
  - Updated `shadow-focus-inset` to gray-60
- Primary
  - Updated `shadow-border-primary` to blue-60
- Destructive
  - Updated `shadow-border-destructive` to red-60
  - Updated `shadow-border-destructive-strong` to red-30
  - Updated `shadow-border-destructive-stronger` to red-20
  - Updated `shadow-border-destructive-strongest` to red-10
- Error
  - Updated `shadow-border-error` to red-60
  - Updated `shadow-border-error-strong` to red-30
- Inverse
  - Removed `color-border-inverse-weaker`
  - Removed `color-border-inverse`
  - Removed `color-border-inverse-strong`
  - Removed `color-border-inverse-stronger`
  - Removed `color-border-inverse-strongest`

### Dark theme: text color
- Updated `color-text-weak` to gray-50
- Updated `color-text-weaker` to gray-80
- Updated `color-text-weakest` to new alias gray-110
- Updated `color-text-inverse-weaker` to gray-70
- Link
  - Updated `color-text-link-weak` to gray-80
  - Updated `color-text-link-stronger` to gray-0
  - Updated `color-text-link-destructive-stronger` to gray-0
- Success
  - Updated `color-text-success` to green-30
- Neutral
  - Updated `color-text-neutral` to blue-20
- New
  - Updated `color-text-new` to purple-20
- Icon
  - Updated `color-text-icon` to gray-50

### CODE REVIEW

File: .changeset/chilled-pears-cough.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+'@twilio-paste/button': patch
+'@twilio-paste/checkbox': patch
+'@twilio-paste/display-pill-group': patch
+'@twilio-paste/input-box': patch
+'@twilio-paste/modal': patch
+'@twilio-paste/radio-group': patch
+'@twilio-paste/skeleton-loader': patch
+'@twilio-paste/toast': patch
+'@twilio-paste/tooltip': patch
+---
```
Comment (SiTaggart) [inline .changeset/chilled-pears-cough.md:16]: Need some words in here

File: packages/paste-design-tokens/tokens/themes/dark/global/border-color.yml [hunk 2]
```diff
@@ -13,107 +13,95 @@ props:
...
-    comment: Border on inverse backgrounds. Must be used on color-background-body-inverse.
-  color-border-inverse-strong:
-    value: "{!palette-gray-0-transparent-70}"
-    comment: Strong border on inverse backgrounds. Must be used on color-background-body-inverse.
-  color-border-inverse-stronger:
-    value: "{!palette-gray-0-transparent-90}"
-    comment: Stronger border on inverse backgrounds. Must be used on color-background-body-inverse.
-  color-border-inverse-strongest:
-    value: "{!palette-gray-0}"
-    comment: Strongest border on inverse backgrounds. Must be used on color-background-body-inverse.
```
Comment (SiTaggart) [inline packages/paste-design-tokens/tokens/themes/dark/global/border-color.yml:35]: Just making double sure, these now inherit from the default theme, correct?

Comment (loreina) [reply packages/paste-design-tokens/tokens/themes/dark/global/border-color.yml:35]: Yeah! Inverse border tokens are now the same in Default and Dark theme

File: .changeset/thick-bags-thank.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/paste-website': patch
+'@twilio-paste/core': patch
+---
+
+[paste-website] Update navigation sidebar right border to use `colorBorderWeak`
```
Comment (shleewhite) [inline .changeset/thick-bags-thank.md:6]: You don't need a changeset for the website

File: .changeset/metal-parents-crash.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+'@twilio-paste/design-tokens': minor
+'@twilio-paste/core': minor
+'@twilio-paste/color-contrast-utils': patch
+---
+
+[design-tokens] Update alias and add new tokens to be accessible in Dark theme
+[color-contrast-utils] Enable color contrast tests for Dark theme
```
Comment (shleewhite) [inline .changeset/metal-parents-crash.md:8]: color contrast utils should probably be separated into its own changeset

File: packages/paste-core/components/radio-group/src/Radio.tsx [hunk 5]
```diff
@@ -86,11 +86,16 @@ const Radio = React.forwardRef<HTMLInputElement, RadioProps>(
           >
             <Box
               as="span"
-              backgroundColor={state.disabled && state.checked ? 'colorBackgroundStrongest' : 'colorBackgroundBody'}
-              borderRadius="borderRadiusCircle"
-              height="sizeSquare25"
-              width="sizeSquare25"
-            />
+              element={`${element}_CONTROL_CIRCLE`}
```
Comment (shleewhite) [inline packages/paste-core/components/radio-group/src/Radio.tsx:89]: Is this something we want to let people customize?

File: packages/paste-core/components/display-pill-group/src/DisplayPill.tsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ export const DisplayPill = React.forwardRef<HTMLAnchorElement, DisplayPillProps>
           alignItems="center"
           backgroundColor="colorBackgroundStrong"
           borderRadius="borderRadius10"
-          color={props.href ? 'colorTextLinkStrong' : 'colorText'}
```
[the remark this answers is not in the document]
Comment (loreina) [reply packages/paste-core/components/display-pill-group/src/DisplayPill.tsx:38]: We want it to use `colorText` for both cases. The underline under the text when it's a `href` will indicate it's an Anchor and mirrors how we have it in Badge. We can revisit visual changes for Pill when there's a link when we do stage 2

File: .changeset/shiny-lions-call.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+---
+'@twilio-paste/alert': patch
```
Comment (andipants12) [inline .changeset/shiny-lions-call.md:2]: Are these minor or patch? Asking because I think this results in a visual change, value changes for the tokens?

File: packages/paste-color-contrast-utils/__tests__/themeContrast.spec.ts [hunk 8]
```diff
@@ -156,7 +156,7 @@ describe('Default Theme', () => {
   });
 });
 
-describe.skip('Dark Theme', () => {
```
Comment (andipants12) [inline packages/paste-color-contrast-utils/__tests__/themeContrast.spec.ts:159]: nice!

File: packages/paste-core/components/radio-group/src/Radio.tsx (hunk 5 above)
Comment (loreina) [reply packages/paste-core/components/radio-group/src/Radio.tsx:89]: We let them customize the control so we could leave it? I'm not familiar with what this line allows users to do though. @TheSisb would love some help here since we just copied this from `IconWrapper` I believe

Comment (andipants12) [reply packages/paste-core/components/radio-group/src/Radio.tsx:89]: Would love some additional context about this change please. ☝️

Comment (andipants12) [reply packages/paste-core/components/radio-group/src/Radio.tsx:89]: Also, could we please add a test for this? Feel free to message me and I can help if needed, we follow a pretty consistent pattern for all the customization `element` prop tests, so there are examples in the component packages.

**Review by andipants12 — COMMENTED** [review]
🙌 Left some comments around test coverage and clarification on the added `element` prop to the `<Radio />` component. Looks great though!

File: .changeset/shiny-lions-call.md (hunk 7 above)
Comment (loreina) [reply .changeset/shiny-lions-call.md:2]: These are just visual changes, we swapped some of the tokens for other ones. Would that be minor or patch?

Comment (SiTaggart) [reply .changeset/shiny-lions-call.md:2]: Patch for me. We're just tweaking existing things visually. There's no extra functionality or changes in functionality. 

Radio might be the only one that's minor as it has a new element you can customize.

File: packages/paste-core/components/radio-group/src/Radio.tsx (hunk 5 above)
Comment (SiTaggart) [reply packages/paste-core/components/radio-group/src/Radio.tsx:89]: FWIW we allow a consumer to customize the checkmark in a checkbox. The dot in a radio button is essentially the same thing, so it makes sense to me to have a similar level customization between both checkbox and radio.

+1 on adding a test though, because as soon as we add a customizable element, we must ensure it stays.

**Review by SiTaggart — APPROVED** [review]
Outside of a test and maybe giving Radio its own changeset for a minor calling out the new element, you're good to go from me

File: packages/paste-design-tokens/tokens/themes/dark/global/box-shadow.yml [hunk 9]
```diff
@@ -18,13 +18,13 @@ props:
...
     comment: Shadow for focus ring on interactive elements.
   shadow-focus-inverse:
-    value: "{!offset-0} {!offset-0} {!offset-0} {!offset-20} {!palette-gray-0-transparent-40}"
+    value: "{!offset-0} {!offset-0} {!offset-0} {!offset-20} {!palette-gray-50}"
     comment: Shadow for focus ring on interactive elements on inverse backgrounds.
   shadow-focus-inset:
-    value: "inset {!offset-0} {!offset-0} {!offset-0} {!offset-10} {!palette-gray-0-transparent-50}"
+    value: "inset {!offset-0} {!offset-0} {!offset-0} {!offset-10} {!palette-gray-60}"
     comment: Shadow for inset focus on elements, such as DataGrid cells.
   shadow-border-weaker:
```
Comment (serifluous) [inline packages/paste-design-tokens/tokens/themes/dark/global/box-shadow.yml:29]: Do shadow-border-weak and shadow-border-weaker need to match border colors? (gray-80 and gray-90)

cc @katieporter

File: packages/paste-core/components/checkbox/src/Checkbox.tsx [hunk 10]
```diff
@@ -59,18 +59,16 @@ const CheckboxIcon: React.FC<{
...
-    return <MinusIcon element={element} decorative color={color} size="sizeIcon10" />;
+    return <MinusIcon element={element} decorative color="inherit" size="sizeIcon10" />;
   }
   return (
     <CheckboxCheckIcon
       element={element}
       display={!checked ? 'none' : 'block'}
       decorative
-      color={color}
+      color="inherit"
```
[the remark this answers is not in the document]
Comment (loreina) [reply packages/paste-core/components/checkbox/src/Checkbox.tsx:71]: done

**Review by serifluous — APPROVED** [review]

**Review by TheSisb — APPROVED** [review]
🎉 !

**Comment by loreina** [comment]
Hi there! This is ready for a final review. The last few changes, which are also added to the PR description, to look at:

- Update `palette-gray-110` to new hex value
- Update `shadow-elevation-10` to `black-transparent-40`
- Update `shadow-elevation-20` to `black-transparent-50`
- Update `shadow-elevation-30` to `black-transparent-50`
- Hid `palette-gray-110` from gradient on color foundations page
- Chromatic UI tests: https://www.chromatic.com/build?appId=5e53448165911c0022e68c74&number=3809

**Review by shleewhite — APPROVED** [review]

File: .changeset/metal-parents-crash.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    - Updated `color-text-link-stronger` to gray-0
+    - Updated `color-text-link-destructive-stronger` to gray-0
+  - Success
+    - Updated `color-text-success` to green-30
+  - Neutral
+    - Updated `color-text-neutral` to blue-20
+  - New
+    - Updated `color-text-new` to purple-20
+  - Icon
+    - Updated `color-text-icon` to gray-50
```
Comment (SiTaggart) [inline .changeset/metal-parents-crash.md:147]: Praise: *love* the level of detail in this changeset

File: .changeset/shiny-lions-call.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+- Update animated background color to use `rgba(255, 255, 255, 0.4)`
+
+[toast]
+
+- Update neutral Toast border left color to `colorBorderNeutralWeak`
+
+[tooltip]
+
+- Update tooltip arrow stroke to `colorBorderInverseWeaker`
+- Update tooltip border color to `colorBorderInverseWeaker`
```
Comment (SiTaggart) [inline .changeset/shiny-lions-call.md:69]: praise: super useful context

File: .changeset/yellow-eggs-provide.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+---
+'@twilio-paste/radio-group': minor
+'@twilio-paste/core': minor
+---
+
+[radio-group]
+
+- Add new customization element for the radio control circle
```
Comment (SiTaggart) [inline .changeset/yellow-eggs-provide.md:8]: nitpick (non-blocking): if you wanted to make this _even better_ you might mention what the element is, what it controls and what it's called.
